datafusion/optimizer/src/limit_push_down.rs

```diff
@@ -576,8 +424,8 @@ mod test {
 
         // Should push the limit down to table provider
         // When it has a select
-        let expected = "Limit: skip=10, fetch=1000\
-        \n  Projection: test.a\
+        let expected = "Projection: test.a\
+        \n  Limit: skip=10, fetch=1000\
         \n    TableScan: test, fetch=1010";
```

Comment:
As for `limit-tablescan`.
I'm not sure whether `tablescan` can precise `limit`, so I keep the `limit`.

If `tablescan` limit is completely accurate, we can remove this limit for `limit-tablescan`

Reply:
I think we currently don't expect for tablescan to return exactly at most the limit nr of rows AFAIK.
For this to be possible we need to put some more info into the table scan trait, just like we do with statistics.

Reply:
https://github.com/apache/arrow-datafusion/blob/8c02485cfc3d2c48bc48a557db58e3e5a0f75777/datafusion/core/src/datasource/datasource.rs#L66-L70

Implies that in fact we still need the `Limit` node even when it has been pushed to the scan as well